js2-mode.el

```diff
@@ -6163,8 +6163,6 @@ its relevant fields and puts it into `js2-ti-tokens'."
                              (setf (js2-token-beg token) (- js2-ts-cursor 2))
                              (js2-skip-line)
                              (setf (js2-token-comment-type token) 'line)
-                             ;; include newline so highlighting goes to end of window
```

Comment:
It's semantically meaningful too: `//` comments stretch until the end of the line, so it makes certain amount of sense to include the newline.

And doing it at highlighting time is inconvenient: see the end of `js2-get-token-internal`, special-casing `js2-COMMENT` there is inconvenient, moreso because line and block comments get the same token type.